ci/scripts/cpp_build.sh

```diff
@@ -105,8 +105,6 @@ if [ "${ARROW_EMSCRIPTEN:-OFF}" = "ON" ]; then
     -DCMAKE_C_FLAGS="${CFLAGS:-}" \
     -DCMAKE_CXX_FLAGS="${CXXFLAGS:-}" \
     -DCMAKE_CXX_STANDARD="${CMAKE_CXX_STANDARD:-17}" \
-    -DCMAKE_INSTALL_LIBDIR=${CMAKE_INSTALL_LIBDIR:-lib} \
-    -DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX:-${ARROW_HOME}} \
```

Comment:
Why do we need to remove them?

Reply:
because we want it to install into emscripten sysroot

Reply:
(otherwise it can't find them when building e.g. python or other things that depend on it)